code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -110,6 +128,46 @@
 		adjustBruteLoss(-L.maxHealth/2)
 	L.gib()
 
+/mob/living/simple_animal/hostile/megafauna/proc/handle_phases()
+	switch(phase_transition_style)
+		if(ONE_PHASE) //no extra phases, no reason to switch!
+			return
+		if(PHASE_TRANSITION_HEALTH)
+			if(!phase_transition_threshold.len) //how'd this even happen?
+				WARNING("Megafauna [name] ([x], [y], [z]) tried to check health thresholds for phases without actually having any thresholds defined. Yell at the coders!")
+				phase_transition_style = ONE_PHASE
+				return
+			for(var/V in phase_transition_threshold)
+				var/threshold = phase_transition_threshold[V]
+				if(health / maxHealth <= threshold && "[phase]" != V) //if we're lower than the threshold, swap phases (as long as we're not in that phase already!)
+					phase_transition(text2num(V))
+				else if(health / maxHealth > threshold && phase == text2num(V) && phase_transition_style != PHASE_TRANSITION_HEALTH_NOREGRESS) //we can also regress to old phases!
```

Comment:
I managed to shrink down the bug, and my trail went cold when I was trying to call the actual phase transition proc; it wouldn't work, no matter what I did.